ais/src/main/java/au/gov/amsa/ais/AisMessageParser.java

```diff
@@ -72,10 +80,27 @@ else if (id == 19)
 			return new AisPositionBExtended(message, source, padBits);
 		else if (id == 21)
 			return new AisAidToNavigation(message, source, padBits);
+		else if (id == AisMessageType.STATIC_DATA_REPORT.getId()) {
+			return parseStaticDataReport(id, message, source, padBits);
+		}
 		else if (id == 27)
 			return new AisPositionGPS(message, source, padBits);
 		else
 			return new AisMessageOther(id, source, padBits);
 	}
-
+	
+	private AisMessage parseStaticDataReport(int id, String message, String source, int padBits) {
+		Optional<AisStaticDataReportPart> dataReportPart = 
+				AisStaticDataReportPart.lookup(AbstractAisBStaticDataReport.extractPartNumber(factory, message, padBits));
+		
+		if(dataReportPart.isPresent()) {
+			switch(dataReportPart.get()) {
+				case PART_A: return new AisBStaticDataReportPartA(message, source, padBits);
+				case PART_B: return new AisBStaticDataReportPartB(message, source, padBits);
+				default: return new AisMessageOther(id, source, padBits);
```

Comment:
throw here? I see this in a few places. If we don't find what we expect we should throw right? Isn't the spec clear that if is message 24 then must be Part A or Part B?

Reply:
Do you mean throw an exception? ie a parse exception (AisParseException)?

Reply:
Probably, check other classes to see what our convention is.

Reply:
The parse method that this is called from returns an AisMessageOther if a message type was not identified. I followed the same convention but this may not be exactly the same scenario as this covers messages type parsing that are yet to be implemented whereas this is a A or B. I will change to be a parse exception.

Reply:
Fixed